pyaedt/maxwell.py

```diff
@@ -340,62 +357,63 @@ def setup_ctrlprog(
+                        if line.startswith("work_dir"):
+                            fo.write("work_dir = r'{0}'\n".format(self.working_directory))
+                        elif line.startswith("lib_dir"):
+                            fo.write("lib_dir = r'{0}'\n".format(source_dir))
+                        else:
+                            fo.write(line)
+            else:
+                if file_str is not None:
+                    with io.open(ctl_file_path, "w", newline="\n") as fo:
+                        fo.write(file_str)
+                    assert os.path.exists(ctl_file_path), "Control program file could not be created."
 
-        if aedt_lib_dir:
-            source_dir = aedt_lib_dir
-        else:
-            source_dir = self.pyaedt_dir
-
-        if os.path.exists(ctl_file) and keep_modifications:
-            with open(ctl_file, "r") as fi:
-                existing_data = fi.readlines()
-            with open(ctl_file, "w") as fo:
-                first_line = True
-                for line in existing_data:
-                    if first_line:
-                        first_line = False
-                        if python_interpreter:
-                            fo.write("#!{0}\n".format(python_interpreter))
-                    if line.startswith("work_dir"):
-                        fo.write("work_dir = r'{0}'\n".format(ctl_path))
-                    elif line.startswith("lib_dir"):
-                        fo.write("lib_dir = r'{0}'\n".format(source_dir))
-                    else:
-                        fo.write(line)
+            self.oanalysis.EditSetup(
+                setupname,
+                [
+                    "NAME:" + setupname,
+                    "Enabled:=",
+                    True,
+                    "UseControlProgram:=",
+                    True,
+                    "ControlProgramName:=",
+                    ctl_file_path,
+                    "ControlProgramArg:=",
+                    "",
+                    "CallCtrlProgAfterLastStep:=",
+                    True,
+                ],
+            )
+            return True
         else:
-            if file_str is not None:
-                with io.open(ctl_file, "w", newline="\n") as fo:
-                    fo.write(file_str)
-                assert os.path.exists(ctl_file), "Control program file could not be created."
-
-        self.oanalysis.EditSetup(
-            setupname,
-            [
-                "NAME:" + setupname,
-                "Enabled:=",
-                True,
-                "UseControlProgram:=",
-                True,
-                "ControlProgramName:=",
-                ctl_file_compute,
-                "ControlProgramArg:=",
-                "",
-                "CallCtrlProgAfterLastStep:=",
-                True,
-            ],
-        )
-
-        return True
+            self.logger.error("Control Program is only available in Maxwell 2D and 3D Transient solutions.")
```

Comment:
```suggestion
            self.logger.error("Control program is only available in Maxwell 2D and 3D Transient solutions.")
```

Reply:
sure. I think that solutions and everything that has an equivalent in AEDT UI should be capitalized. So for Transient is should be capital. Thanks